src/frontend/src/components/SearchBar/SearchBar.jsx

```diff
@@ -75,10 +94,17 @@ export default function CustomizedInputBase() {
           placeholder="Search Telescope"
           inputProps={{ 'aria-label': 'search telescope' }}
         />
-        <IconButton type="submit" className={classes.iconButton} aria-label="search">
+        <IconButton
+          type="submit"
+          className={classes.iconButton}
+          aria-label="search"
+          onClick={() => getResults()}
```

Comment:
For debugging, this might be OK, but in general you want to pass your event handlers down from a parent control to a controlled component.  So we'd put the search results logic in the parent of this search component, and then just wire up the button here.

Reply:
you mean like
```
function handleOnClick(){
  useLazyQuery()
}
```

Reply:
Here's a very rough outline of what I mean:

```jsx
// In Parent component, which manages the state for the search, and does the gql stuff...
function ParentComponent() {
  const [searchText, setSearchText] = useState('');
  const onSearchClick = () => { /* do your gql search here... */ };

  return (
    <ChildSearchComponent onSearchClick={onSearchClick} currentSearchText={searchText} />
  );
}

// In Child component
function ChildSearchComponent(props) {
  // Get the click handler, and current text to put in the search bar from the parent...
  const { onSearchClick, currentSearchText } = props;
  return (
    <div>
      <TextInput value={currentSearchText} />
      <Button onClick={onSearchClick}>Search</Button>
    </div>
  );
}
```

You manage the search text, and "on click" handler in a parent component, and the search component just displays buttons and your text, and tells you when to do things (run the function).

Reply:
Think I understand. Once I can actually get this working, I'll set it up like this.